src/tyr/isochrone_serializer.cc

```diff
@@ -85,17 +85,29 @@ serializeIsochrones(const Api& request,
       }));
     }
   }
-  // Add original locations to the geojson
+  // Add input and snapped locations to the geojson
   if (show_locations) {
+    auto snapped_coords = array({});
+    auto input_coords = array({});
     for (const auto& location : request.options().locations()) {
-      features->emplace_back(
-          map({{"type", std::string("Feature")},
-               {"properties", map({})},
-               {"geometry", map({{"type", std::string("Point")},
-                                 {"coordinates", array({fp_t{location.ll().lng(), 6},
-                                                        fp_t{location.ll().lat(), 6}})}})}}));
+      const valhalla::LatLng& latLng_snapped = location.path_edges(0).ll();
```

Comment:
here you actually want another loop over all the path edges. there can be multiple edge candidates per input location depending on the radius you provide. so yeah just add another for loop here over `location.path_edges()`

Reply:
But we don't want to return the `ll()` points for all candidate edges for every origin, right? I wasn't sure what it exactly means if there's multiple edges for an origin, but I figured it's when the starting node is connected to more than one edge. In the few debug tests I did I only ever saw the same `ll()` point per origin for all its candidate edges. But that's not correct then, I guess?!

Also, what exact point on an edge does `ll()` reference? If I remember right, I think I couldn't quite figure that out.

Reply:
i dont see why we wouldnt want to show all of the candidates that were used to make the isochrone. so when you give a lat lon to valhalla, what happens is we "intersect" that with the graphs *edges*. if you specified a radius of 100m, you'll get a bunch of snapped locations to a bunch of edges for a given location. we use *all* of these edges snapped locations to seed the expansion list that computes the isochrones. in other words, its as if you started at all those locations. if you didnt specify a radius you wont get multiple snapped locations except in the case that the closest thing to your input location was a *node* in the graph, in which case you will get a snapped location at that exact node for every edge that leaves and enters it (that is accessable to your mode of travel). i hope that explains it!

Reply:
Thanks for the explanation @kevinkreiser . That explains why I only observed the same snapped locations for multiple candidate edges, I never specified a radius.

From a user perspective, I'd expect a single point per origin point in the case where I can't correlate the center points to origin points, as is the case currently for Valhalla as you pointed out earlier. I just Imagine origins being close together would otherwise potentially have overlapping clusters of center points. Does that make sense?

Reply:
Maybe return the gravity point of center point clusters?